### ISSUE TITLE
subscriptions seems not to support MongoDb ObjectIDs

### ISSUE THREAD

**Issue by apheliagames** [hiddenswitch/Meteor-Unity#25]
just tried to use some MongoDB ObjectId fields in a subscription....

it gets serialized, but the ID itself gets overwritten with "000000000000000000000000".

All fine when i change back to strings, but it would be a nice feature for security reasons...

**Comment by doctorpangloss** [comment]
Yeah they definitely do not support object IDs. I think I'd have to rethink this concept of having things inherit from a MongoDocument...

**Comment by apheliagames** [comment]
Hey Benjamin,

 

Thanks for your fast answer….yeah it would be really cool if we could use native BSON ObjectIds here…

I’m currently using GUIDs instead for referencing other collections ( JSONFX is able to serialize it as Guids if they are strings in the document)  but I can’t use GUIDs as document ids… Meteor is not supporting this, as you surely know…

Keep up  the great work, the rest works great and reliable! 

 

 

Greetz from Berlin
Robert



Aphelia Games
Robert Klausch
Grosse-Leege Str. 97

13155 Berlin / Germany

++49 30 224 673 - 11

 

 

Von: Benjamin Berman [mailto:notifications@github.com] 
Gesendet: Montag, 22. Mai 2017 22:31
An: hiddenswitch/Meteor-Unity
Cc: apheliagames; Author
Betreff: Re: [hiddenswitch/Meteor-Unity] subscriptions should support MongoDb ObjectIDs serialization (#25)

 

Yeah they definitely do not support object IDs. I think I'd have to rethink this concept of having things inherit from a MongoDocument...
